File: reimi/frontend/reimi_app/lib/domain/value_objects/annual_income.dart
import 'package:freezed_annotation/freezed_annotation.dart';

enum AnnualIncome {
  @JsonValue('UNDER_2M')
  under2M,
  @JsonValue('BETWEEN_2M_TO_4M')
  between2And4M,
  @JsonValue('BETWEEN_4M_TO_6M')
  between4And6M,
  @JsonValue('BETWEEN_6M_TO_8M')
  between6And8M,
  @JsonValue('BETWEEN_8M_TO_10M')
  between8And10M,
  @JsonValue('BETWEEN_10M_TO_15M')
  between10And15M,
  @JsonValue('BETWEEN_15M_TO_20M')
  between15And20M,
  @JsonValue('BETWEEN_20M_TO_30M')
  between20And30M,
  @JsonValue('OVER_30M')
  over30M,
}

File: reimi/backend/reimi_app/src/main/java/com/reimi/reimi_app/domain/model/profile/AnnualIncome.java
package com.reimi.reimi_app.domain.model.profile;

public enum AnnualIncome {
    UNDER_2M("200万円未満"),
    BETWEEN_2M_TO_4M("200万~400万円"),
    BETWEEN_4M_TO_6M("400万~600万円"),
    BETWEEN_6M_TO_8M("600万~800万円"),
    BETWEEN_8M_TO_10M("800万~1000万円"),
    BETWEEN_10M_TO_15M("1000万~1500万円"),
    BETWEEN_15M_TO_20M("1500万~2000万円"),
    BETWEEN_20M_TO_30M("2000万~3000万円"),
    OVER_30M("3000万円以上");

    private final String label;

    AnnualIncome(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}
